src/NuGet.Core/NuGet.Protocol/CachingSourceProvider.cs

```diff
@@ -24,18 +24,24 @@ public class CachingSourceProvider : ISourceRepositoryProvider
         private readonly ConcurrentDictionary<string, SourceRepository> _cachedSources
             = new ConcurrentDictionary<string, SourceRepository>(StringComparer.Ordinal);
 
-        public CachingSourceProvider(IPackageSourceProvider packageSourceProvider)
+        public CachingSourceProvider(IPackageSourceProvider packageSourceProvider,
+            IEnumerable<Lazy<INuGetResourceProvider>> resourceProviders)
         {
             _packageSourceProvider = packageSourceProvider;
```

Comment:
I think we should consider adding some basic tests for this class & some null checks.